File: Python/Bubble Sort/sort.py
def bubbleSort(array):
	n = len(array)
	rval = list(array)
	if n == 0:
		status = False
		return [status, rval]
	swapped = True
	while swapped == True:
		swapped = False
		for i in range(1,n,1):
			if rval[i-1] > rval[i]:
				old = rval[i]
				rval[i] = rval[i-1]
				rval[i-1] = old
				swapped = True
	status = True
	return [status, rval]

File: C/Bubble Sort/sort.c
int bubbleSort(int *array, int size);
int swap(int *a,int *b);

int swap(int *a,int *b) {
	int old = *a;
	*a = *b;
	*b = old;
	return 0;
}
int bubbleSort(int *array, int size) {
	int swapped = 1;
	int i = 0;
	int r = 0;
	while(swapped == 1) {
		swapped = 0;
		for(i=1;i<size;i++) {
			if(array[i-1] > array[i]) {
				r = swap(&array[i-1],&array[i]);
				swapped = 1;
			}
		}
	}
	return 0;
}
